echopype/calibrate/api.py

```diff
@@ -22,6 +19,51 @@
 logger = _init_logger(__name__)
 
 
+def _check_echodata_backscatter_size(echodata: EchoData):
```

Comment:
We didn't cover this in our discussion, but I think we can move this function to be under `CalibrateBase`. The reason is that, for EK80 data where some channels transmit in CW mode and some channels transmit BB (LFM) mode, the data size can be dramatically different between the 2 beam groups, because BB mode data _must be_ stored in complex format (whereas CW mode data can be stored in either power/angle or complex format). So if you sum up the size across the 2 beam groups it may be misleading, since the CW mode data could actually be pretty small and won't cause any problems.

You can find a class attribute `ed_beam_group` in `CalibrateEK`, which will give you the right beam group to access for the calibration. So you could just grab the combined size of `backscatter_*` of that beam group. 

AZFP does not have this complication, so the combined size of `backscatter_r` would suffice.

Reply:
@leewujung Thanks, this is very helpful! So here's the cases that I think we have for EK80 (from some digging through the data test suite):

BB and Complex -> Grab `ed_beam_group` and calculate sum nbytes in `backscatter_i` and `backscatter_r` and these variables should have a `beam` dimension
CW and Complex -> Grab `ed_beam_group` and calculate sum nbytes in `backscatter_i` and `backscatter_r`and these variables should have a `beam` dimension
CW and Power -> Grab `ed_beam_group` and calculate sum nbytes in `backscatter_r` and this variable should not have a `beam` dimension

Reply:
And for EK60 and AZFP, I should not expect to use anything other than `Sonar/Beam_group1` right?

Reply:
EK60 and AZFP will only have `Sonar/Beam_group1`. I think our docs is clear about that. If not please propose changes.